plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,226 @@
+package io.jenkins.plugins.casc.misc;
+
+import com.gargoylesoftware.htmlunit.WebRequest;
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.util.NameValuePair;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.logging.Level;
+import org.apache.commons.io.IOUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import static com.gargoylesoftware.htmlunit.HttpMethod.POST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Base test to check a complete test of each plugin configuration. What it makes:
+ * 1.  Configure the instance with the {@link #configResource()} implemented.
+ * 2.  Check it was configured correctly.
+ * 3.  Export the full Jenkins configuration via Web UI (hacked as it fails so far).
+ * 4.  Export the schema via Web UI (commented out as there is no good validator so far).
+ * 5.  Verify the Jenkins configuration against the schema (commented out as there is no good validator so far).
+ * 6.  Check the Jenkins configuration is valid via Web UI (used the plugin config so far).
+ * 7.  Apply the Jenkins configuration via Web UI (maybe not needed, but we test it as well).
+ * 8.  Write the Jenkins configuration to $JENKINS_ROOT/jenkins.yaml.
+ * 9.  Restart Jenkins.
+ * 10. Check the {@link #stringInLogExpected()} is set during the restart.
+ *
+ * All the plugin author needs to do is override the methods providing:
+ * 1. The resource with the yaml configuration of the plugin
```

Comment:
Now it's optional and it's not mandatory to override
```suggestion
 * 1. The resource with the yaml configuration of the plugin in case they use their own name for the file
```